superbench/common/utils/topo_aware.py

```diff
@@ -35,37 +35,24 @@ def search(self, pattern, string, flags=0):
         return self.matched
 
 
-def gen_ibstat_file(ibstat_file):
-    """Generate ibstat file for each node with specified path.
+def gen_ibstat_file(host_list, ibstat_file):
+    """Generate ibstat file in each node with specified path.
 
     Args:
+        host_list (list): list of VM read from hostfile.
         ibstat_file (str): path of ibstat output.
     """
-    from mpi4py import MPI
-
-    if not MPI.Is_initialized():
-        MPI.Init()
-
-    comm = MPI.COMM_WORLD
-    name = MPI.Get_processor_name()
-
-    # The command to fetch ibstat info
-    cmd = r"ibstat | grep -Po 'System image GUID: \K\S+$'"
-    output = os.popen(cmd)
-    ibstat = 'VM_hostname ' + name + '\n' + str(output.read())
-
-    # Fetch all ibstate from each node
-    ibstats = comm.allgather(ibstat)
-
-    ibstate_file_path = Path(ibstat_file)
-
-    # Filter the duplicate info
-    ibstat_infos = set(ibstats)
-
-    with ibstate_file_path.open(mode='w') as f:
-        for ibstat_info in ibstat_infos:
-            f.write(ibstat_info)
-    MPI.Finalize()
+    try:
+        pssh_cmd = "pssh -i -t 5 -H '{}' ".format(' '.join(host_list))
+        cmd = "'cat /sys/class/infiniband/*/sys_image_guid" \
+            r"| tr -d :' | sed 's/^.*[[[:upper:]]\{7\}]/VM_hostname/g' | cut -d ' ' -f 1,2"
```

Comment:
change [[:upper:]]{7} to (SUCCESS|FAILURE)? there're only two options, you may need sed -E

Reply:
addressed